lib/finder.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+class Finder
+  def initialize(user_interface, database)
+    @user_interface = user_interface
+    @database = database
+  end
+
+  def run
+    loop do
+      search_result = database.search(user_interface.search_term)
+      display_results(search_result)
```

Comment:
this method name `display_results` is misleading, here `search_result` is the result of a database search, but you don't always get a result back, so you may not have results to display. Only if you actually got results you would display them. 

Something like `handle_result` (singular) would be more general and include all cases. It implies you may or may not get results and but are going to handle each case appropriately anyway.